content/en/docs/instrumentation/java/manual.md

```diff
@@ -7,31 +7,198 @@ aliases:
+
+public class Dice {
+
+  private int min;
+  private int max;
+
+  public Dice(int min, int max) {
+    this.min = min;
+    this.max = max;
+  }
+
+  public List<Integer> rollTheDice(int rolls) {
+    List<Integer> results = new ArrayList<Integer>();
+    for (int i = 0; i < rolls; i++) {
+      results.add(this.rollOnce());
+    }
+    return results;
+  }
+
+  private int rollOnce() {
+    return ThreadLocalRandom.current().nextInt(this.min, this.max + 1);
+  }
+}
+```
+
+Create the app files `DiceApplication.java` and `RollController.java` and add
+the following code to them:
+
+```java
+// DiceApplication.java
+package otel;
 
-### Maven
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.Banner;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class DiceApplication {
+  public static void main(String[] args) {
+
+    SpringApplication app = new SpringApplication(DiceApplication.class);
+    app.setBannerMode(Banner.Mode.OFF);
+    app.run(args);
+  }
+}
+```
+
+```java
+// RollController.java
+package otel;
+
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import otel.Dice;
+
+@RestController
+public class RollController {
+  private static final Logger logger = LoggerFactory.getLogger(RollController.class);
+
+  @GetMapping("/rolldice")
+  public List<Integer> index(@RequestParam("player") Optional<String> player,
+      @RequestParam("rolls") Optional<Integer> rolls) {
+
+    if (!rolls.isPresent()) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Missing rolls parameter", null);
+    }
+
+    List<Integer> result = new Dice(1, 6).rollTheDice(rolls.get());
+
+    if (player.isPresent()) {
+      logger.info("{} is rolling the dice: {}", player.get(), result);
+    } else {
+      logger.info("Anonymous player is rolling the dice: {}", result);
+    }
+    return result;
+  }
+}
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```shell
+java -jar ./build/libs/java-simple.jar
+```
+
+You should get a list of 12 numbers in your browser window, for example:
+
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
+```
+
+## Manual instrumentation setup
+
+For both library and app instrumentation, the first step is to install the
+dependencies for the OpenTelemetry API.
+
+<!-- prettier-ignore-start -->
+
+{{< tabpane text=true >}}
+{{% tab Gradle %}}
+
+```kotlin
+dependencies {
+    implementation 'io.opentelemetry:opentelemetry-api:{{% param javaVersion %}}'
```

Comment:
This could be slightly confusing to inexperienced users who might replace their ENTIRE dependencies block. I think it should be additive since it builds on the previous block.
```suggestion
    implementation("org.springframework.boot:spring-boot-starter-web")
    implementation 'io.opentelemetry:opentelemetry-api:{{% param javaVersion %}}'
```

Reply:
This might also be true of other gradle dependency blocks below.

Reply:
good call out, I reduced the list to make it more obvious what is new, but maybe [this](https://gohugo.io/content-management/syntax-highlighting/#highlighting-in-code-fences) can help

Reply:
Yeah I'd be totally happy with that. Thanks @svrnm

Reply:
updated and I used `hl_lines` to highlight the lines of new dependencies and code changes overall

Reply:
good improvement, thanks!